cmake/env/thisbdm.sh

```diff
@@ -32,355 +32,401 @@ drop_bdm_from_path()
    local p=$1
    local drop=$2
 
-   newpath=`echo $p | sed -e "s;:${drop}:;:;g" \
-                          -e "s;:${drop}\$;;g"   \
-                          -e "s;^${drop}:;;g"   \
-                          -e "s;^${drop}\$;;g"`
+   newpath=$(echo "$p" | sed -e "s;:${drop}:;:;g"\
+                             -e "s;:${drop}\$;;g"\
+                             -e "s;^${drop}:;;g" \
+                             -e "s;^${drop}\$;;g")
 }
 
-if [ -n "${BDM_INSTALL_DIR}" ] ; then
-   old_bdmsys_base=${BDM_INSTALL_DIR}
-fi
-if [ -n "${BDMSYS}" ] ; then
-   old_bdmsys=${BDMSYS}
-fi
+source_thisbdm()
+{
+  # detect bash-like shell
+  local bdm_shell
```

Comment:
You can use $SHELL, which is set to the shell type already. See https://github.com/Homebrew/install/blob/master/install.sh#L636